monitor/drfeedback-monitor.py

```diff
@@ -0,0 +1,235 @@
+                self.vprint ('Average number of drfeedback alerts is {} over last {} samples'.format(
+                    average_alerts, self.args['--samples']))
+
+                if not last_average_value:
+                    last_average_value = average_alerts
+                else:
+
+                    if self._detect_spike(average_alerts, last_average_value):
+                        self.vprint ('A spike was detected! current-average={} last-average={} ' \
+                                     'threshold={} max-permitted-average={}'.format(
+                                         average_alerts, last_average_value,
+                                         self.args['--threshold'], average_alerts * self.args['--threshold']))
+                        return True
+                    else:
+                        last_average_value = average_alerts
+
+        self.vprint('no spike detected')
+        return False
+
+    def _detect_spike(self, average_alerts, last_average):
+        '''
+        Returns True if spike is detected
+        '''
+        self.vprint ('Spike check: current-average={} last-average={} threshold={} max-permitted-average={}'.format(
+                        average_alerts, last_average, self.args['--threshold'], average_alerts * self.args['--threshold']))
+        
+        is_spike = average_alerts * self.args['--threshold'] < last_average
+        if is_spike:
+            self.vprint ('Spike was detected! {} * {} < {}'.format(average_alerts, self.args['--threshold'], last_average))
+
+        return is_spike
+
+    def slack_alert(self, simulate):
+        '''
+        Sends a network slack alert to the Dev Team to bring attentio to DrFeedback logs
+        '''
+
+        url = self.url_simulate if simulate else self.url_slack
+        self.vprint('Slack alert signal requested, simulate={} url={}'.format(simulate, url))
+
+        command = 'curl -X POST --data-urlencode \'payload=' \
+                  '{"channel": "#team-os", "username": "drfeedback", ' \
+                  '"text": "DrFeedback Monitor detected a spike: http://dev.kano.me/feedback-reports", ' \
+                  '"icon_emoji": ":face_with_monocle:"}\' ' + url
+
+        self.vprint(command)
+
+        # FIXME: Find a reliable way to test it in simulation mode
+        if not simulate:
+            os.system(command)
+
+        return
+
+    def run(self):
+        '''
+        Run is the main entry point to put this class task to work
+        '''
+        counters = self._get_log_counters_()
+        return self._compute_(counters)
+
+        
+
+
+
+if __name__ == '__main__':
+
+    DRFEEDBACK_DIRECTORY = '/var/cache'
+    LOGFILE = 'drfeedback-numbers.txt'
+    SAMPLES = 5
+    THRESHOLD = 1.5
+
+    args = docopt.docopt(__doc__)
+    rc = 0
+
+    if args['--verbose']:
+        print args
+    
+    # Provide default values when not specified
+    if not args['--directory']:
+        args['--directory'] = DRFEEDBACK_DIRECTORY
```

Comment:
You can do defaults entirely in Docopt if you like:

```
Usage:
    --directory=<pathname>    Directory to search for logs [default: /some/dir]
```

Look at the end of [the documentation](http://docopt.org/) for details.

You can combine this with the `DRFEEDBACK_DIRECTORY` (and other variables) with something like:

```python
DRFEEDBACK_DIRECTORY = '/var/cache'
LOGFILE = 'drfeedback-numbers.txt'
SAMPLES = 5
THRESHOLD = 1.5
USAGE = """
DRFeedback-monitor keeps track and alerts the Kano Team about potential Kano OS customer troubles

Usage:
    drfeedback-monitor collect --logfile=<filename> --directory=<pathname> [--commit]
    drfeedback-monitor process --logfile=<filename> [--alert] [--samples=<n>] [--threshold=<factor>] [--verbose]

Options:
    -h, --help                   Show this message.
    -c, --commit                 Write the new collected counter to the logfile
    -a, --alert                  Send a slack alert if spike is detected
    -V, --verbose                Explain the steps taken during the process
    --directory=<pathname>       Directory to search for logs [default: {log_dir}]
    --logfile=<filename>         File to store the logs to [default: {log_file}]
    --samples=<n>                Number of samples to use [default: {samples}]
    --threshold=<factor>         Threshold for alerts [default: {threshold}]
   
""".format(
    log_dir=DRFEEDBACK_DIRECTORY,
    log_file=LOGFILE,
    samples=SAMPLES,
    threshold=THRESHOLD
)


...
def main():
    args = docopt.docopt(USAGE)
    ...
```

Reply:
Not resolved, can't replace variable args for some reason